### DISCUSSION TITLE
xDrip+Variant & (BYO) Dexcom G6 app

### DISCUSSION THREAD

**Discussion by AliQ101** [NightscoutFoundation/xDrip#2174] [Q&A]
I use two xDrip+ apps on my mobile with the following data source:
- xDrip+ (latest stable version 31.1.22) with data source: xDrip+Sync Follower  (for my child with Libre/MiaoMiao)
- xDrip+Variant (latest nightly build) with data source: Dex Share Follower  (for me with Dexcom G6)
I have also installed BYO Dexcom app (as mobile does not work with the original Dexcom app) with no broadcasting.
That works fine.
When I have installed BYO Dexcom app with local broadcasting to xDrip+ and changed xDrip+Variant data source to: 640G/EverSense then for next 3-4 readings all was great (Xdrip+Variant displayed my direct G6 readings and xDrip+ displayed Libre readings). Then from next data reading my G6 reading was displayed in both xdrip+Variant and xDrip+ (with unchanged data source). This continues until I have uninstalled this BYO Dexcom app, then xDrip+ switched back to show Libre data from next reading automatically and xDrip+Variant had no reading (as no G6 app installed). There are no special loggs in xDrip+, when showed G6 readings, no LibreBlock value created in that time.
I am not sure whether the issues in with BYO Dexcom app (it does not distinguish between xDrip+/xDrip+Variant app? - but why first 3-4 readings are ok) or is there any local broadcasting between the both xDrip+/xDrip+Variant apps (but all options for local broadcasting are turn off) or is it issue with data source +Sync Follower? 
Any ideas?

**Comment by Navid200** [comment]
Temporarily disable the main xDrip collection from xDrip sync.
Set the variant data source to 640G / EverSense.  Does it work consistently?

Now, enable the main xDrip to collect from xDrip sync.  Does the variant stop working?
If yes, disable the main xDrip source again.  Does the variant work again?

**Comment by AliQ101** [comment]
Thanks for your comments, in meantime I made more testing:
Installed BYO Dex app with broadcasting first, then:
Step 1)  xDrip source sync, Variant source Share, Dex app share is disabled = all works fine (Variant log: "jamorham nsemulator - received NSEmulater data but we are not a follower or emulator receiver")
Step 2) xDrip source sync, Variant source Share, Dex app share abled = all works fine (Variant log: the same as above)
Step 3)  xDrip source sync, Variant source 640G/EverSense, Dex app share disabled = in first reading both xDrip and Variant displayed G6 value, then after second reading xDrip showed Libre and Variant showed G6 value (Variant log: "jamorham nsemulator" disappeared).  It worked fine also for next readings. Only small issue: time in xDrip widget between readings was always reset to 0 when Variant made reading, but no effect to bg readings.

Then I turned off wifi and data on my mobile for testing:
4) next reading xDrip and Variant displayed G6 value (xDrip log: LibreBlock was created, but bg table contains - raw:85 NSEmulator follower instead of only e.g. raw:142)
I turn on wifi and mobile data back:
5) nothing has changed, still both xDrip and Variant displayed G6 value.
6) disabled data source in Variant, xDrip still source sync = no readings in Variant but xDrip continue to display G6 value (log: LibreBlock was created...)
7) disabled data source in xDrip, Variant source 640G/EverSense = no readings in xDrip and G6 values in Variant. Then I changed xDrip data source back to sync and Libre data was shown again. Variant displayed G6 value. So all works fine again. Further, issue with time in widget disappeared.

I tried to turn off wifi/data again = back to point 4)
It seems to me that issues is within xDrip/Variant based on point 4 and 6.

UPDATE for step/point 3 above:
- it works fine only for a random period of time (sometimes for 10 hours during night, sometimes only for half hour in day), definitely is not stable
- any treatment (e.g. carbs) made in xDrip (Libre follower) and sent to master = back to point 4)
